File: src/main/java/ch/zxseitz/tbsg/repo/IRepository.java
package ch.zxseitz.tbsg.repo;

import org.bson.types.ObjectId;

import java.util.List;
import java.util.Optional;

public interface IRepository<T> {
    void insert(T t);
    List<T> getAll();
    Optional<T> get(ObjectId id);
    void update(ObjectId id, T t);
    void delete(ObjectId id);
}

File: server/src/main/java/ch/zxseitz/tbsg/server/repo/IRepository.kt
package ch.zxseitz.tbsg.server.repo;

import org.bson.types.ObjectId

interface IRepository<T> {
    fun insert(t: T)
    fun getAll(): List<T>
    fun get(id: ObjectId): T?
    fun update(id: ObjectId, t: T);
    fun delete(id: ObjectId);
}
